fix: disclosure levels

timcadman opened the pull request:
## What's changed
- Added checks for disclosure mode which were missing from certain functions
- Added section in readme detailing which functions are permitted in which disclosure mode
- Added additional checks to block functions that could be used in inference attacks in conjuncture with mutate

StuartWheater reviewed (commented):
- Should the package version be 1.0.3?
- You could include in the "permissions level" table for a row for 'non-permissive'

StuartWheater commented:
Appears to conflicts need resolving:

**Conflicting files**
- docs/authors.html
- docs/pkgdown.yml

timcadman commented:
> * Should the package version be 1.0.3?
> * You could include in the "permissions level" table for a row for 'non-permissive'

The package version will be updated automatically on merge with master

timcadman commented:
> * Should the package version be 1.0.3?
> * You could include in the "permissions level" table for a row for 'non-permissive'

Updated permissions table, including row for non-permissive and also updated which function is in which category

StuartWheater commented:
Should ".getEncodeKey()" [R/utils.R line 20] be ".getEncodeDictionary"?

timcadman commented:
> Should ".getEncodeKey()" [R/utils.R line 20] be ".getEncodeDictionary"?

Yes - I've just correct, thanks
